arangod/Graph/Providers/ClusterProvider.cpp

```diff
@@ -378,12 +397,25 @@ auto ClusterProvider<StepImpl>::fetch(std::vector<Step*> const& looseEnds)
 
   if (!looseEnds.empty()) {
     result.reserve(looseEnds.size());
-    fetchVerticesFromEngines(looseEnds, result);
-    _stats.addHttpRequests(_opts.engines()->size() * looseEnds.size());
+    if (!_opts.produceVertices()) {
```

Comment:
This is good for now, but for future we could get this a bit more clever, and figure out which vertices we actually need.